### PR TITLE
Initial Commit for Steps Component

### PR DESCRIPTION
**Pull request by Parag27-cakewalk**
https://github.com/CakewalkUAE/softstudio-projectmanagement/issues/468

### CODE REVIEW

File: apps/www/registry/tui/example/cards/steps.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
+import { Card, CardContent, CardHeader, CardTitle, } from "@/registry/tui/ui/card"
+import { Steps } from "../../ui/steps";
+
+export function CardsSteps() {
```
Comment (raghavcakewalk) [inline apps/www/registry/tui/example/cards/steps.tsx:4]: please use arrow functions everywhere

File: apps/www/registry/tui/example/cards/steps.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    const steps = [
+        { id: "Step 1", name: "Job details", borderClass: "border-indigo-600", textClass: "text-indigo-600",status: 'complete' },
+        { id: "Step 2", name: "Application form", borderClass: "border-indigo-600", textClass: "text-indigo-600", status: 'current' },
+        { id: "Step 3", name: "Preview", borderClass: "border-gray-200", textClass: "text-gray-500", status: 'upcoming' }
+    ];
+
+    const stepsData = [
```
Comment (raghavcakewalk) [inline apps/www/registry/tui/example/cards/steps.tsx:11]: memoize constant data

File: apps/www/registry/tui/ui/steps.tsx [hunk 3]
```diff
@@ -0,0 +1,416 @@
...
+                                                            </div>
+                                                        </>
+                                                    ) : null}
+                                                </li>
+                                            ))}
+                                        </ol> : null
+                }
+
+            </nav>
+        </>
```
Comment (raghavcakewalk) [inline apps/www/registry/tui/ui/steps.tsx:412]: <> </> not required

File: apps/www/registry/tui/ui/steps.tsx [hunk 4]
```diff
@@ -0,0 +1,416 @@
+import * as React from "react"
+import { cva, type VariantProps } from "class-variance-authority"
+import { cn } from "@/lib/utils"
+import { Icon, IconType } from "./icon"
+import { colors } from "./helper/types"
+
+const stepsVariants = cva(
```
Comment (raghavcakewalk) [inline apps/www/registry/tui/ui/steps.tsx:7]: sure it should be empty?

**Review by raghavcakewalk — CHANGES_REQUESTED** [review]
changes requested

File: apps/www/registry/tui/ui/steps.tsx [hunk 5]
```diff
@@ -0,0 +1,410 @@
...
+    borderColor?:colors;
+    anchorSize?: "default" | "xl";
+    spanSize?: "default" | "m";
+    divSize?: "default" | "l" | "m" | "xl";
+
+}
+function classNames(...classes: string[]) {
+    return classes.filter(Boolean).join(' ')
+}
+function Steps({ children, className, iconStyle, backgroundColor,borderColor, textColor, spanSize, divSize, steps, simple, anchorSize, bulletOfText, bulletStepText, panels, bullets, circle, bulletText, progressBar, icon, circleText, ...props }: StepsProps) {
```
Comment (raghavcakewalk) [inline apps/www/registry/tui/ui/steps.tsx:113]: arrow function please

File: apps/www/registry/tui/example/cards/steps.tsx [hunk 6]
```diff
@@ -0,0 +1,87 @@
+import { Card, CardContent, CardHeader, CardTitle, } from "@/registry/tui/ui/card"
+import { Steps } from "../../ui/steps";
+import { useMemo } from "react";
+
+export const CardsSteps = ()=> {
+    const steps = [
```
Comment (raghavcakewalk) [inline apps/www/registry/tui/example/cards/steps.tsx:6]: useMemo for this also

**Review by shubhenducw — APPROVED** [review]

**Review by kaushal248 — APPROVED** [review]

**Pull request merged**